### PR TITLE
fix: not update `starDifficultyBindable` immediately in `computeStarRating()` to avoid flicker

### LINKED ISSUE #36964: Beatmap flicker when using mods & importing new maps
### Type

Cosmetic

### Bug description

Whenever a new beatmap is imported, I imagine with mods selected (didn't test without selecting any), the map card flashes and the difficulties flicker

### Screenshots or videos

https://github.com/user-attachments/assets/1d4e949f-7c0a-4520-acbb-60a02509ead7

### Version

2026.305.0-lazer

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/25987690/compressed-logs.zip)

### PR DESCRIPTION
**Pull request by hesitling**
- When modifying Mod/Ruleset, the difficulty cache update causes the beatmap difficulty value to change. At this point, the updated value is calculated with Mod and Ruleset applied (the correct new value).
- During `PrepareForUse()`, it binds to the difficulty cache update, and because `runOnceImmediately = true`, it immediately triggers an update. At this moment, both the old value and the updated value are the beatmap's base difficulty values (no Mod, no Ruleset).
- The difficulty cache updates again. The old value is the base value, and the new value is the difficulty value calculated with Mod and Ruleset applied.

The flickering occurs because the intermediate `PrepareForUse()` reverts the difficulty display value back to the beatmap's base difficulty value. Setting `runOnceImmediately = false` resolves the difficulty value flickering issue.

Resolves #36964.

### CODE REVIEW

File: osu.Game/Screens/Select/PanelBeatmap.cs
Comment (peppy) [inline osu.Game/Screens/Select/PanelBeatmap.cs:246]: It feels as though there could be a threading edge case where the value is updated between these two lines, but the threaded retrieval *does* schedule back to `Update` thread for the actual updates, so it should be okay.

I'd kinda hope to see test coverage though.

**Comment by hesitling** [comment]
Maybe another solution.

```c#
--- a/osu.Game/Beatmaps/BeatmapDifficultyCache.cs
+++ b/osu.Game/Beatmaps/BeatmapDifficultyCache.cs
@@ -131,10 +131,20 @@ public void Invalidate(IBeatmapInfo oldBeatmap, IBeatmapInfo newBeatmap)
         /// <returns>A bindable that is updated to contain the star difficulty when it becomes available. May be an approximation while in an initial calculating state.</returns>
         public IBindable<StarDifficulty> GetBindableDifficulty(IBeatmapInfo beatmapInfo, CancellationToken cancellationToken = default, int computationDelay = 0)
         {
+            var difficulty = new StarDifficulty(beatmapInfo.StarRating, 0);
+
+            var localBeatmapInfo = beatmapInfo as BeatmapInfo;
+
+            if (localBeatmapInfo != null)
+            {
+                var lookup = new DifficultyCacheLookup(localBeatmapInfo, currentRuleset.Value, currentMods.Value);
+                if (CheckExists(lookup, out var cached) && cached != null)
+                    difficulty = cached.Value;
+            }
             var bindable = new BindableStarDifficulty(beatmapInfo, cancellationToken)
             {
                 // Start with an approximate known value instead of zero.
-                Value = new StarDifficulty(beatmapInfo.StarRating, 0)
+                Value = difficulty
             };

             updateBindable(bindable, currentRuleset.Value, currentMods.Value, cancellationToken, computationDelay);
```
